lib/CTownHandler.cpp

```diff
@@ -987,11 +967,13 @@ CFaction * CTownHandler::loadFromJson(const std::string & scope, const JsonNode
 	auto preferUndergound = source["preferUndergroundPlacement"];
 	faction->preferUndergroundPlacement = preferUndergound.isNull() ? false : preferUndergound.Bool();
 
-	//Contructor is not called here, but operator=
-	auto nativeTerrain = source["nativeTerrain"];
-	faction->nativeTerrain = nativeTerrain.isNull()
-		? getDefaultTerrainForAlignment(faction->alignment)
-		: VLC->terrainTypeHandler->getInfoByName(nativeTerrain.String())->id;
+	faction->nativeTerrain = ETerrainId::NONE;

```

Comment:
No more getDefaultTerrainForAlignment?

Reply:
Correct. Problem is, that code would also trigger for faction of neutral creatures from H3 who are not supposed to have any native terrain, as results Nomads/Azure's/Snipers/etc got native terrain they were not supposed to have.